src/main/java/org/jboss/ejb/client/DiscoveryEJBClientInterceptor.java

```diff
@@ -512,7 +513,10 @@ Discovery getDiscovery() {
                         }
                         final String nodeName = nodeValue.toString();
                         nodes.put(location, nodeName);
-                        uris.put(nodeName, location);
+                        // we want to prioritize using local invocation
+                        if(!uris.containsKey(nodeName) || location.equals(LOCAL_NODE_NAME)) {
+                            uris.put(nodeName, location);
+                        }
```

Comment:
`location` is URI, `LOCAL_NODE_NAME` is String, so this expression will always evaluate to false.

Reply:
this was a bug, fixed